kernel/src/main/scala/cats/kernel/Eq.scala

```diff
@@ -76,6 +76,14 @@ object Eq extends EqFunctions[Eq] with EqToEquivConversion {
       def eqv(x: A, y: A) = eq1.eqv(x, y) || eq2.eqv(x, y)
     }
 
+  /**
+    * Return an Eq that gives the result of conjunction of all arguments
+    * Note this is idempotent
+    */
+  def allOf[@sp A](eqs: Eq[A]*): Eq[A] =
```

Comment:
I think that you are just following what's done in similar methods here, but I question whether specialization is likely to help us here. It seems unlikely to me that someone is going to define multiple definitions of equivalence for a primitive type and what to check that all of them are satisfied. I suspect that this would only really serve to increase the cats jar size. WDYT @johnynek?